src/mbgl/renderer/renderer_impl.cpp

```diff
@@ -185,6 +186,10 @@ void Renderer::Impl::render(const UpdateParameters& updateParameters) {
 
         if (layerAdded || layerChanged) {
             layer.transition(transitionParameters);
+
+            if (layer.is<RenderHeatmapLayer>()) {
+                layer.as<RenderHeatmapLayer>()->updateColorRamp();
+            }
```

Comment:
This means we're updating the color ramp every time a heatmap-color layer changes, even if the change was to some other property.  Should we check the diff more granularly here?